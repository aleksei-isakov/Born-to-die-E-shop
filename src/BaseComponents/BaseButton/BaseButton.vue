<template>
  <button class="BaseButton" @click="click">
    <slot></slot>
  </button>
</template>

<script>
export default {
  name: 'BaseButton',

  props: {},

  data() {
    return {};
  },

  methods: {
    click() {
      this.$emit('click', data);
    }
  }
};
</script>
