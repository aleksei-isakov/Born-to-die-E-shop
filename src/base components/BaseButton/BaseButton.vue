<template>
  <button class="BaseButton" v-on:click="click">
    <slot></slot>
  </button>
</template>

<script>
export default {
  name: 'BaseButton',

  props: {},

  methods: {
    onClickSendDataToParent() {
      this.$emit('click');
    }
  }
};
</script>
