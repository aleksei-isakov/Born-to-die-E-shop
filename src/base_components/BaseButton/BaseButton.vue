<template>
  <button class="base-button" @click="sendDataToParent">
    <slot></slot>
  </button>
</template>

<script>
export default {
  name: 'base-button',
  props: {},
  data() {
    return {};
  },
  methods: {
    sendDataToParent() {
      this.$emit('sendDataToParent', data);
    }
  }
};
</script>

<style lang="scss"></style>
