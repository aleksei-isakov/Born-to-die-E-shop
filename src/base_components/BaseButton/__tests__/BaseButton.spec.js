import { shallowMount } from '@vue/test-utils';
import BaseButton from '../BaseButton.vue';

describe('BaseButton.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(BaseButton, {
      methods: {
        sendDataToParent() {
          this.$emit('sendDataToParent', data);
        }
      }
    });
    expect(wrapper.find('button')).toBeTruthy();
    expect(wrapper.is(BaseButton)).toBeTruthy();
  });
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
