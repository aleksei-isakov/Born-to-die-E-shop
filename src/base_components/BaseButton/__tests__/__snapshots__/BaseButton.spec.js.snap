// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton.vue has the expected html structure 1`] = `
<button
  class="base-button"
/>
`;
